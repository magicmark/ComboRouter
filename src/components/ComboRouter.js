// @flow

import * as React from 'react';
import { StaticRouter, BrowserRouter } from 'react-router-dom';

const IS_BROWSER = () => typeof window !== 'undefined' && window.document && window.document.createElement;

// eslint-disable-next-line no-unused-vars
const ChosenRouter = IS_BROWSER ? BrowserRouter : StaticRouter;

type Props = {|
  basename: string,
  children: ?React.Node,
  staticRouterProps?: Object,
  browserRouterProps?: Object,
|};

const ComboRouter = (props: Props) => {
  const passedProps = IS_BROWSER ? props.browserRouterProps : props.staticRouterProps;
  return <ChosenRouter {...passedProps}>{props.children}</ChosenRouter>;
};

export default ComboRouter;
