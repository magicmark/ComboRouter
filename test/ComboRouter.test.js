import React from 'react';
import { render } from 'react-testing-library';
import ComboRouter from '../src/components/ComboRouter';

// smoke test
it('ComboRouter renders correctly', () => {
  const { container } = render(<ComboRouter><p>foo</p></ComboRouter>);
  expect(container.firstChild).toMatchSnapshot();
});
