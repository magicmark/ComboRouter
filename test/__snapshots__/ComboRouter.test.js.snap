// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComboRouter renders correctly 1`] = `
<p>
  foo
</p>
`;
